### DISCUSSION TITLE
Error while creating staging models using Scalefree for dbt

### DISCUSSION THREAD

**Discussion by ssuri39** [ScalefreeCOM/datavault4dbt#129] [Q&A]
I am getting below  error while compiling a staging model using scalefree package . Please suggest what is wrong with the code. Issue seems while specifying the source model but need to know how to fix.The database is snowflake .

**stg_sample** is a model that just selects all data from the source table in landing schema in snowflake.

**Compilation Error in model hash_stg_sample (models/staging/hash_stg_sample.sql)
The name argument to ref() must be a string, got <class 'dbt.clients.jinja.create_undefined.<locals>.Undefined'>**
 

**Code** :

    {{ config(materialized='view') }}

    {%- set yaml_metadata -%}
    source_model: 'stg_sample'
    ldts: 'current_timestamp'
    rsrc: '!dummy'
    hashed_columns: 
        hk_hub:
            - col1
            - col2
        hd_sat:
            is_hashdiff: true
            columns:
                - sat1
                - sat2
                - sat3
                - sat4
    
    {%- endset -%}

    {%- set metadata_dict = fromyaml(yaml_metadata) -%}

    {%- set source_model = metadata_dict['source_model'] -%}
    {%- set ldts = metadata_dict['ldts'] -%}
    {%- set rsrc = metadata_dict['rsrc'] -%}
    {%- set hashed_columns = metadata_dict['hashed_columns'] -%}

    {{ datavault4dbt.stage(source_model=source_model,
                        ldts=ldts,
                        rsrc=rsrc,
                        hashed_columns=hashed_columns) }}

**Comment by bschlottfeldt** [comment]
Hi @ssuri39 it seems like a yml compilation problem.
Could you try removing the newline located after the` - sat4` and before `{%- endset -%}`. Let me know if it works.

**Reply by ssuri39** [reply]
Sorry but it didn't work. have attached the file after the change for your review. thanks
[sample_stage_scalefree.txt](https://github.com/ScalefreeCOM/datavault4dbt/files/12737210/sample_stage_scalefree.txt)

**Reply by bschlottfeldt** [reply]
How is your stg_sample defined, could you post here the code for this source model ? And is it located in the models folder ?
Try checking if you have any other dbt models that could have a syntax yml error, dbt wont let you compile any files if one of them has a compilation problem.
Kind regards,
Barbara

**Reply by ssuri39** [reply]
<img width="683" alt="stage_sample_scalefree" src="https://github.com/ScalefreeCOM/datavault4dbt/assets/146161033/f2aefa35-8e7a-4543-a15a-ba95e4ccd55c">
 

I have attached the snippet of the stg_sample model and the directory its in is also visible. It is in models folder and there are no other dbt models with compilation issue. This stg_sample is also running ok.

Thanks for the help and please let me know if any other info is needed!

**Reply by bschlottfeldt** [reply]
Hi @ssuri39 i got an email from someone with a similar issue, this is how he described the problem which resulted in the same error you got, and the solution:

"I already managed to get this fixed. The issue was that after copy paste there was one line space in the beginning of each row. This caused the condition for source model reference in the new line to be incorrectly used. Aligning all the code to the left fixed the problem. 

Potentially that person from that Wiki could have a similar issue but it is hard to tell this."
Maybe this is also the reason for the error you get, but it could be different..
Let me know if it works,
Kind regards,
Barbara
